graql/reasoner/state/CompositeState.java

```diff
@@ -94,7 +93,10 @@ public ResolutionState propagateAnswer(AnswerState state) {
                 .noneMatch(q -> q.resolve(subGoals, true).findFirst().isPresent());
 
         return isNegationSatisfied?
-                new AnswerState(answer, getUnifier(), getParentState()) :
-                null;
+                new AnswerState(
+                        new ConceptMap(answer.map(), answer.explanation(), getQuery().getPattern(answer.map())),
```

Comment:
hopefully this doesn't have any implications with performance, rewriting the concept maps more often :) we can bench in taxfix